### PR TITLE
added casts and unused vars to remove warnings

### PR DESCRIPTION
**Pull request by CrackedPixel**
No more warnings when building =]

### CODE REVIEW

File: src/raygui.h [hunk 1]
```diff
@@ -1280,7 +1280,7 @@ static const char *guiTooltipPtr = NULL;        // Tooltip string pointer (strin
 static bool guiSliderDragging = false;          // Gui slider drag state (no inputs processed except dragged slider)
 static Rectangle guiSliderActive = { 0 };       // Gui slider active bounds rectangle, used as an unique identifier
 
-static int textBoxCursorIndex = 0;     // Cursor index, shared by all GuiTextBox*()
+static int textBoxCursorIndex = 0;              // Cursor index, shared by all GuiTextBox*()
```
Comment (CrackedPixel) [inline src/raygui.h:1283]: this was previously `unsigned` and when it got changed the indenting for the comment got moved

File: src/raygui.h [hunk 2]
```diff
@@ -1612,7 +1612,7 @@ int GuiTabBar(Rectangle bounds, const char **text, int count, int *active)
     #define RAYGUI_TABBAR_ITEM_WIDTH    160
 
     int result = -1;
-    GuiState state = guiState;
+//    GuiState state = guiState;
```
Comment (CrackedPixel) [inline src/raygui.h:1615]: i prefer to delete unused code, but i saw a pattern of commenting them out instead (such as on line #1480, #3409, etc)

Comment (raysan5) [reply src/raygui.h:1615]: By raylib/raygui conventions, comments for unused variable are always in fornt of the variable, with no spaces.

File: src/raygui.h [hunk 3]
```diff
@@ -1527,9 +1527,9 @@ int GuiGroupBox(Rectangle bounds, const char *text)
 
     // Draw control
     //--------------------------------------------------------------------
-    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x, bounds.y, RAYGUI_GROUPBOX_LINE_THICK, bounds.height }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BORDER_COLOR_DISABLED : LINE_COLOR)));
-    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x, bounds.y + bounds.height - 1, bounds.width, RAYGUI_GROUPBOX_LINE_THICK }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BORDER_COLOR_DISABLED : LINE_COLOR)));
-    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x + bounds.width - 1, bounds.y, RAYGUI_GROUPBOX_LINE_THICK, bounds.height }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? BORDER_COLOR_DISABLED : LINE_COLOR)));
+    GuiDrawRectangle(RAYGUI_CLITERAL(Rectangle){ bounds.x, bounds.y, RAYGUI_GROUPBOX_LINE_THICK, bounds.height }, 0, BLANK, GetColor(GuiGetStyle(DEFAULT, (state == STATE_DISABLED)? (int)BORDER_COLOR_DISABLED : (int)LINE_COLOR)));
```
Comment (raysan5) [inline src/raygui.h:1530]: I prefer to avoid those casts, considering it is not done anywhere else for those defined values.

**Comment by raysan5** [comment]
I'm not merging this PR, the warning are actually not that dramatic.

**Pull request closed without merging**